fractal_server/app/routes/api/v1/project.py

```diff
@@ -430,6 +431,37 @@ async def apply_workflow(
         ),
         **apply_workflow.dict(),
     )
+
+    # Rate Limiting:
+    # raise `429 TOO MANY REQUESTS` if this endpoint has been called with the
+    # same database keys (Project, Workflow and Datasets) during the last
+    # `settings.FRACTAL_API_SUBMIT_RATE_LIMIT` seconds.
+    stm = (
+        select(ApplyWorkflow)
+        .where(ApplyWorkflow.project_id == project_id)
+        .where(ApplyWorkflow.workflow_id == workflow_id)
+        .where(ApplyWorkflow.input_dataset_id == input_dataset_id)
+        .where(ApplyWorkflow.output_dataset_id == output_dataset_id)
+        .order_by(ApplyWorkflow.start_timestamp.desc())
+        .limit(1)
```

Comment:
Could you explain how the `order_by` and `limit` options work together? If we are sorting by timestamp, why would we only keep one of the output items?

Reply:
Of all the jobs with these foreign keys, the only one we need is the most recent one, so taking others would be redundant.